### PR TITLE
Merge in Next to Main.

### PR DESCRIPTION
**Pull request by i3149**
Includes meraki update, #334 , #335.

### CODE REVIEW

File: pkg/inputs/snmp/x/meraki/meraki.go [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+		return nil, err
+	}
+
+	// Now, load latency for any of these which have them:
+	err = c.getUplinkLatencyLoss(dur, uplinkSet)
+	if err != nil {
+		return nil, err
+	}
+
+	//c.log.Infof("Loaded %d Uplinks for %d Organizations", len(uplinkSet), len(c.orgs))
```
Comment (ehazlett) [inline pkg/inputs/snmp/x/meraki/meraki.go:516]: can we just remove?

File: pkg/inputs/snmp/x/meraki/meraki.go [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+			"recent_device_name": client.RecentDeviceName,
+			"dhcp_hostname":      client.DhcpHostname,
+			"mdsn_name":          client.MdnsName,
+		}
+		dst.CustomInt = map[string]int32{
+			"vlan": int32(client.Vlan),
+		}
+		dst.CustomBigInt = map[string]int64{}
+		dst.EventType = kt.KENTIK_EVENT_SNMP_DEV_METRIC
+		//dst.Provider = c.conf.Provider // @TODO, pick a provider for this one.
```
Comment (ehazlett) [inline pkg/inputs/snmp/x/meraki/meraki.go:320]: do we need to have a provider?

File: pkg/inputs/snmp/x/meraki/meraki.go [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+		return nil, err
+	}
+
+	//c.log.Infof("Loaded %d Uplinks for %d Organizations", len(uplinkSet), len(c.orgs))
+
+	return c.parseUplinks(uplinkSet)
+}
+
+type uplinkTS struct {
+	TS          time.Time `json:"ts":`
```
Comment (ehazlett) [inline pkg/inputs/snmp/x/meraki/meraki.go:522]: typo?  should it be `json:"ts"` (no extra colon).

File: pkg/inputs/snmp/x/meraki/meraki.go [hunk 4]
```diff
@@ -0,0 +1,685 @@
...
+				"interface":       uplink.Interface,
+				"provider":        uplink.Provider,
+				"signal_type":     uplink.SignalType,
+				"signal_rsrp":     uplink.SignalStat.Rsrp,
+				"signal_rsrq":     uplink.SignalStat.Rsrq,
+			}
+			dst.CustomInt = map[string]int32{}
+			dst.CustomBigInt = map[string]int64{}
+			dst.EventType = kt.KENTIK_EVENT_SNMP_DEV_METRIC
+			//dst.Provider = c.conf.Provider // @TODO, pick a provider for this one.
```
Comment (ehazlett) [inline pkg/inputs/snmp/x/meraki/meraki.go:638]: need a provider?

File: pkg/kt/azure.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+func loadViaAzureKeyVault(key string) string {
+	if azureClient == nil {
+		keyVaultName := os.Getenv("KT_AZURE_KEY_VAULT_NAME") // Must be set.
+		if keyVaultName == "" {
+			log.Printf("ENV Var 'KT_AZURE_KEY_VAULT_NAME' must be set")
+			return "ENV Var KT_AZURE_KEY_VAULT_NAME must be set"
+		}
+
+		keyVaultURL := fmt.Sprintf("https://%s.vault.azure.net/", keyVaultName) // Should this be hard coded?
```
Comment (ehazlett) [inline pkg/kt/azure.go:36]: Looks like these can possibly have different private external endpoints (https://docs.microsoft.com/en-us/azure/key-vault/general/private-link-service?tabs=portal) but probably fine for now.

File: pkg/inputs/snmp/x/meraki/meraki.go (hunk 1 above)
Comment (i3149) [reply pkg/inputs/snmp/x/meraki/meraki.go:516]: Good call.

File: pkg/inputs/snmp/x/meraki/meraki.go (hunk 2 above)
Comment (i3149) [reply pkg/inputs/snmp/x/meraki/meraki.go:320]: This is a NR specific thing. They don't right now know what provider to set so they asked to remove this for now entirely.

File: pkg/inputs/snmp/x/meraki/meraki.go (hunk 3 above)
Comment (i3149) [reply pkg/inputs/snmp/x/meraki/meraki.go:522]: good catch!

File: pkg/kt/azure.go (hunk 5 above)
Comment (i3149) [reply pkg/kt/azure.go]: Updated to allow a custom endpoint if you really want one.

**Review by ehazlett — APPROVED** [review]

**Pull request merged**